raiden/tests/integration/test_balance_proof_check.py

```diff
@@ -0,0 +1,145 @@
+from raiden.transfer.state_change import ContractReceiveChannelSettled
+
+
+@pytest.mark.parametrize('deposit', [10])
+@pytest.mark.parametrize('channels_per_node', [CHAIN])
+@pytest.mark.parametrize('number_of_nodes', [2])
+def test_invalid_close(
+        raiden_network,
+        number_of_nodes,
+        deposit,
+        token_addresses,
+        network_wait,
+):
+    app0, app1 = raiden_network
+    token_address = token_addresses[0]
+    chain_state = views.state_from_app(app0)
+    payment_network_id = app0.raiden.default_registry.address
+    registry_address = app0.raiden.default_registry.address
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        chain_state,
+        payment_network_id,
+        token_address,
+    )
+    channel_identifier = get_channelstate(app0, app1, token_network_identifier).identifier
+
+    # make a transfer
+    direct_transfer(
+        app0,
+        app1,
+        token_network_identifier,
+        1,
+        timeout=network_wait * number_of_nodes,
+    )
+    # stop app1 - the test uses token_network_contract now
+    app1.stop()
+    token_network_contract = TokenNetwork(app1.raiden.chain.client, token_network_identifier)
+
+    # app1 closes the channel
+    token_network_contract.close(
+        channel_identifier,
+        app0.raiden.address,
+        EMPTY_HASH,
+        0,
+        EMPTY_HASH,
+        EMPTY_SIGNATURE,
+    )
+    waiting.wait_for_close(
+        app0.raiden,
+        registry_address,
+        token_address,
+        [channel_identifier],
+        app0.raiden.alarm.sleep_time,
+    )
+    waiting.wait_for_settle(
+        app0.raiden,
+        registry_address,
+        token_address,
+        [channel_identifier],
+        app0.raiden.alarm.sleep_time,
+    )
+    state_changes = app0.raiden.wal.storage.get_statechanges_by_identifier(
+        from_identifier=0,
+        to_identifier='latest',
+    )
+    assert must_contain_entry(state_changes, ContractReceiveChannelSettled, {
+        'token_network_identifier': token_network_identifier,
+        'channel_identifier': channel_identifier,
+    })
+
+
+@pytest.mark.parametrize('deposit', [10])
+@pytest.mark.parametrize('channels_per_node', [CHAIN])
+@pytest.mark.parametrize('number_of_nodes', [2])
+def test_invalid_update_transfer(
+        raiden_network,
+        number_of_nodes,
+        deposit,
+        token_addresses,
+        network_wait,
+        chain_id,
+):
+    app0, app1 = raiden_network
+    token_address = token_addresses[0]
+    chain_state = views.state_from_app(app0)
+    payment_network_id = app0.raiden.default_registry.address
+    registry_address = app0.raiden.default_registry.address
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        chain_state,
+        payment_network_id,
+        token_address,
+    )
+    channel_identifier = get_channelstate(app0, app1, token_network_identifier).identifier
+
+    # make a transfer
```

Comment:
Same comments as in the above test apply.

- Add kwargs in the function calls
- Expand comments a bit so that the future readers easily understand what's happening